controller/controller_test.go

```diff
@@ -270,508 +277,815 @@ func TestUnhealthyIfUpdaterFails(t *testing.T) {
+			Path:                  ingressPath,
+			ServiceAddress:        serviceIP,
+			ServicePort:           ingressSvcPort,
+			LbScheme:              "internal",
+			Allow:                 []string{},
+			StripPaths:            false,
+			BackendTimeoutSeconds: backendTimeout,
+			IngressName:           "test",
+		}},
+		Config{
+			DefaultAllow:                 ingressDefaultAllow,
+			DefaultBackendTimeoutSeconds: backendTimeout,
+			IngressName:                  "test",
 		},
-		{
-			"ingress without path definition",
-			createIngressesFixture(ingressHost, ingressSvcName, ingressSvcPort,
-				map[string]string{}, ""),
-			createServiceFixture(ingressSvcName, "lalala land", serviceIP),
-			nil,
-			defaultConfig(),
+	})
+}
+
+func TestUpdaterIsUpdatedForIngressInNamespaceMatchingSelector(t *testing.T) {
+	runAndAssertUpdates(t, expectGetIngresses, testSpec{
+		"ingress is in a namespace that matches the namespace selector",
+		createIngressesFixture(ingressNamespace, ingressHost, ingressSvcName, ingressSvcPort, map[string]string{
+			ingressAllowAnnotation:      "",
+			stripPathAnnotation:         "false",
+			backendTimeoutSeconds:       "10",
+			frontendElbSchemeAnnotation: "internal",
+			ingressNameAnnotation:       defaultIngressName,
+		}, ingressPath),
+		createDefaultServices(),
+		createNamespaceFixture(ingressNamespace, map[string]string{"team": "theteam"}),
+		[]IngressEntry{{
+			Namespace:             ingressNamespace,
+			Name:                  ingressName,
+			Host:                  ingressHost,
+			Path:                  ingressPath,
+			ServiceAddress:        serviceIP,
+			ServicePort:           ingressSvcPort,
+			LbScheme:              "internal",
+			Allow:                 []string{},
+			StripPaths:            false,
+			BackendTimeoutSeconds: backendTimeout,
+			IngressName:           defaultIngressName,
+		}},
+		Config{
+			DefaultAllow:                 ingressDefaultAllow,
+			DefaultBackendTimeoutSeconds: backendTimeout,
+			IngressName:                  defaultIngressName,
+			NamespaceSelector:            &k8s.NamespaceSelector{LabelName: "team", LabelValue: "theteam"},
 		},
-		{
-			"ingress without rules definition",
-			createIngressWithoutRules(),
-			createServiceFixture(ingressSvcName, "lalala land", serviceIP),
-			nil,
-			defaultConfig(),
+	})
+}
+
+func TestUpdaterIsNotUpdatedForIngressInNamespaceMatchingSelector(t *testing.T) {
+	runAndAssertUpdates(t, expectGetIngresses, testSpec{
+		"ingress is in a namespace that matches the namespace selector",
+		createIngressesFixture(ingressNamespace, ingressHost, ingressSvcName, ingressSvcPort, map[string]string{
+			ingressAllowAnnotation:      "",
+			stripPathAnnotation:         "false",
+			backendTimeoutSeconds:       "10",
+			frontendElbSchemeAnnotation: "internal",
+			ingressNameAnnotation:       defaultIngressName,
+		}, ingressPath),
+		nil,
+		createNamespaceFixture(ingressNamespace, map[string]string{"team": "otherteam"}),
+		nil,
```

Comment:
Expect no entries if the ingress is in a namespace without the matching label selector